### PR TITLE
Enable custom gradients SavedModel option in test

### PR DESCRIPTION
**Pull request by k-w-w**
TensorFlow SavedModel added the option to save custom gradients in [e6c7ba](https://github.com/tensorflow/tensorflow/commit/e6c7ba3875634d7c142d787c2c5012cea9ad8daa). This change updates the JAX SavedModel test to use this option.

### CODE REVIEW

File: jax/experimental/jax2tf/tests/savedmodel_test.py [hunk 1]
```diff
@@ -94,13 +97,8 @@ def f_jax_jvp(primals, tangents):
...
       y = restored_model.f(xv)
-
-    # TODO: at the moment TF does not fully-support custom_gradient in a
-    # savedmodel (b/123499169), but at least a warning is printed and an
-    # exception is thrown when gradients are taken. The exception, however,
-    # is a very strange one, for now.
-    with self.assertRaisesRegex(TypeError, "An op outside of the function building code is being passed"):
-      _ = tape.gradient(y, xv)
+    self.assertAllClose(tape.gradient(y, xv).numpy(),
+                        jax.grad(f_jax)(0.7).astype(np.float32))
```
Comment (gnecula) [inline jax/experimental/jax2tf/tests/savedmodel_test.py:101]: We should use here `x` in lieu of `0.7`.

**Review by gnecula — COMMENTED** [review]
Thank you so much for fixing the custom_gradient support. It has been a long-standing issue for us. 

I would recommend revising also the text in https://github.com/google/jax/blob/3ec5bc503139985c8d099f2ba8bdb71fc059d051/jax/experimental/jax2tf/README.md#no-savedmodel-fine-tuning. I would say instead that we support saving only for the first-order gradient computation.

File: jax/experimental/jax2tf/tests/savedmodel_test.py (hunk 1 above)
Comment (k-w-w) [reply jax/experimental/jax2tf/tests/savedmodel_test.py]: Updated the `x` and documentation

**Review by gnecula — APPROVED** [review]
LGTM. Thanks!
